src/SAWScript/Crucible/LLVM/X86.hs

```diff
@@ -186,8 +186,15 @@ makeLenses ''X86State
 runX86Sim :: X86State -> X86Sim a -> IO (a, X86State)
 runX86Sim st m = runStateT (unX86Sim m) st
 
-throwX86 :: MonadThrow m => String -> m a
-throwX86 = throw . X86Error
+-- | Throw an X86Error (general failure of the X86 stuff) with the
+--   pathname of the file we choked on.
+throwX86 :: MonadThrow m => FilePath -> String -> m a
+throwX86 path msg = throw $ X86Error path Nothing msg
+
+-- | Throw an X86Error (general failure of the X86 stuff) with the
+--   pathname of the file we choked on and also the affected function name.
+throwX86' :: MonadThrow m => FilePath -> String -> String -> m a
+throwX86' path func msg = throw $ X86Error path (Just func) msg
```

Comment:
Perhaps this should be named `throwX86Func` to make it more obvious that the exception requires a particular function name?

Reply:
Sure. Done in ac7ecb3943784f624c619d78d09dc6a0a7ab0474.